src/modules/rlm_sql/rlm_sql.c

```diff
@@ -478,6 +481,28 @@ static size_t sql_escape_func(UNUSED REQUEST *request, char *out, size_t outlen,
 	return len;
 }
 
+/** Passed as the escape function to map_proc and sql xlat methods
+ *
+ * The variant reserves a connection for the escape functions to use, and releases it after
+ * escaping is complete.
+ */
+static size_t sql_escape_for_xlat_func(REQUEST *request, char *out, size_t outlen, char const *in, void *arg)
+{
+	size_t			ret;
+	rlm_sql_t		*inst = arg;
+	rlm_sql_handle_t	*handle;
```

Comment:
Can you switch this out to ``handle = talloc_get_type_abort(arg, rlm_sql_handle_t)`` to make extra sure that argument is correct.